packages/databricks-vscode/src/configuration/configureWorkspaceWizard.ts

```diff
@@ -160,66 +176,187 @@ export async function configureWorkspaceWizard(
+            // TODO: If the profile has an auth type with deeper support in the Extension, (azure-cli, databricks-cli),
+            // then run the necessary checks to validate the profile and help user fix any errors.
+            const authProvider = new ProfileAuthProvider(
+                this.state.host!,
+                pick.profile
+            );
+            const checkResult = await this.checkAuthProvider(
+                authProvider,
+                `profile '${pick.profile}'`
+            );
+            if (checkResult) {
+                return checkResult;
+            }
+            this.state.authProvider = authProvider;
+            return;
+        }
+
+        return (input: MultiStepInput) => this.createNewProfile(input, pick);
+    }
+
+    private async createNewProfile(
+        input: MultiStepInput,
+        pick: AuthTypeQuickPickItem
+    ) {
+        let initialValue = this.configModel.target ?? "";
+
+        // If the initialValue profile already exists, then create a unique name.
+        const profiles = await listProfiles(this.cliWrapper);
+        if (profiles.find((profile) => profile.name === initialValue)) {
+            const suffix = randomUUID().slice(0, 8);
+            initialValue = `${this.configModel.target}-${suffix}`;
+        }
 
-            case "profile":
-                state.authType = pick.authType;
-                state.profile = pick.profile;
+        const profileName = await input.showInputBox({
+            title: "Enter a name for the new profile",
+            step: 3,
+            totalSteps: 3,
+            placeholder: "Enter a name for the new profile",
+            initialValue,
+            validate: async (value) => {
+                if (value.length === 0) {
+                    return {
+                        message: "Profile name cannot be empty",
+                        type: "error",
+                    };
+                }
+                const profiles = await listProfiles(this.cliWrapper);
+                if (profiles.find((profile) => profile.name === value)) {
+                    return {
+                        message: `Profile ${value} already exists`,
+                        type: "error",
+                    };
+                }
+            },
+            ignoreFocusOut: true,
+        });
+
+        if (profileName === undefined) {
+            return;
+        }
+
+        let authProvider: AzureCliAuthProvider | DatabricksCliAuthProvider;
+        switch (pick.authType) {
+            case "azure-cli":
+                authProvider = new AzureCliAuthProvider(this.state.host!);
                 break;
 
-            case "new-profile":
-                await commands.executeCommand(
-                    "databricks.connection.openDatabricksConfigFile"
+            default:
+                // This is the only other case possible right now. We are not explicitly checking for databricks-cli
+                // to make typescript happy.
```

Comment:
Having a new case condition with the `databricks-cli` and throwing exception / logging in the default case would be more useful. Otherwise we might silently do the wrong thing here. Can we improve the types to make it work?